aws-sdk-bench/src/main.rs

```diff
@@ -0,0 +1,104 @@
+use std::error::Error;
+use std::sync::{Arc, Mutex};
+
+extern crate jemallocator;
+
+#[global_allocator]
+static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
+
+use aws_sdk_s3 as s3;
+use s3::Region;
+
+use tracing_subscriber::fmt::format::FmtSpan;
+use tracing_subscriber::fmt::SubscriberBuilder;
+
+use structopt::StructOpt;
+
+fn take_job(to_do: &Mutex<u32>) -> bool {
+    let mut jobs_left = to_do.lock().unwrap();
+    if *jobs_left <= 0 {
+        false
+    } else {
+        *jobs_left -= 1;
+        true
+    }
+}
+
+async fn do_downloads(num_workers: u32, samples: u32, bucket: &str, key_prefix: &str, region: &str) -> Option<&str> {
+    let to_do = Arc::new(Mutex::new(samples));
+    let mut jobs = vec![];
+    let start = std::time::Instant::now();
+
+    // XXX: Logs/tracer might affect perf against the other two impls?
+    SubscriberBuilder::default()
+        .with_env_filter("info")
+        .with_span_events(FmtSpan::CLOSE)
+        .init();
+    let conf = s3::Config::builder()
+        .region(Region::new(region))
+        .build();
+
+    // XXX: Does the client need to be within an Arc?
```

Comment:
client doesn't need to be in an arc, there is an internal arc

Reply:
Thanks @rcoh, true that!